### ISSUE TITLE
RetroArch on Android crashes sporadically when playing Chrono Trigger (FF Chronicles edition) on PCSX-ReARMed core

### ISSUE THREAD

**Issue by gnafuthegreat** [libretro/RetroArch#5844]
## Description

Sporadic crashes while playing the FF Chronicles edition of Chrono Trigger using the PCSX-ReARMed core on Android (both an LG G3 running Lineage OS and a Fire TV with stock firmware).

### Expected behavior

Play without crashes, or at least store the latest save to the memory card and display a helpful error when the core crashes.

### Actual behavior

RetroArch crashes, not just the core, and this has been happening since I started using it in the 1.4 series (but still in the 1.6.9 release).  When it crashes, any saves to the memory card since the last load are lost, but save states are retained (so I've taken to saving the state quite frequently).  This seems to happen most often when switching between rooms in dungeons, but it can really happen at any point.  I haven't played other games much yet, so I don't know if this is a broader issue than Chrono Trigger.

### Steps to reproduce the bug

1. Start a new game in FF Chronicles edition of Chrono Trigger (my copy was ripped using cdrdao, tried both with and without including subchannel data, with both .toc and .cue)
2. Please for a little while
3. It will eventually crash (early in the game, I believe I saw it most when trying to save in-game)

### Bisect Results

I've only been using RetroArch with any regularity in recent 1.4 and 1.6 releases, always with the official build from the Play Store (or in the case of the Fire TV, only 1.6.9 downloaded from the RetroArch website).  I've attached logcat output from the Fire TV shortly after a crash.  Please let me know if there are any other useful logs I should collect, and whether logs from the LG G3 would be helpful.

### Version/Commit

- RetroArch: 1.6.9
- PCSX-ReARMed: r22 c7dde5e

### Environment information

- OS: Lineage OS 14.1 and Fire OS (first gen. Fire TV, latest update)

[retroarch-pcsxrearmed-chrono-trigger-fire-tv.log](https://github.com/libretro/RetroArch/files/1523418/retroarch-pcsxrearmed-chrono-trigger-fire-tv.log)

**Comment by gnafuthegreat** [comment]
I should probably also note that this is with the US/NTSC release of Chrono Trigger.

**Comment by meepingsnesroms** [comment]
This is the important part:
I/RetroArch( 5285): Loading state: "/storage/emulated/0/RetroArch/states/chrono_trigger.state".
I/RetroArch( 5285): State size: 4456448 bytes.
I/PCSX    ( 5285): NI fffffcd5 @a0010000 (a000ef24)
I/PCSX    ( 5285): NI ff7ffffd @a0010004 (a000ef24)
I/PCSX    ( 5285): NI ffffffbf @a0010010 (a000ef24)
I/PCSX    ( 5285): NI 6fc079ff @a0010014 (a000ef24)
I/PCSX    ( 5285): NI ffffffff @a0010018 (a000ef24)
I/PCSX    ( 5285): NI 7fff7fff @a001001c (a000ef24)
I/PCSX    ( 5285): NI 5802950f @a0010054 (a000ef24)
I/PCSX    ( 5285): Compile at bogus memory address: ff000000
I/WindowState( 1832): WIN DEATH: Window{18ce050c u0 com.retroarch/com.retroarch.browser.retroactivity.RetroActivityFuture}
I/ActivityManager( 1832): Process com.retroarch (pid 5285) has died

This is chrono trigger trying to run an invalid address, the ps1 equivalent of SIGSEGV.
Sometimes old games rely on hardware bugs, like trying to execute IO registers or write to ROM.
Are you using the original ps1 bios?

**Comment by gnafuthegreat** [comment]
I am using the scph5501.bin BIOS I've seen recommended in most places (rather than extracted from my own PlayStation; I've not attempted that).  Would it be helpful for me to try it with any other particular BIOS version?  I also haven't tried the HLE BIOS to see how that behaves.

**Comment by meepingsnesroms** [comment]
No the bios is correct, I was just checking because the game might be trying to run from the bios in a weird way.
Does this only happen when you load that state?

**Comment by gnafuthegreat** [comment]
I've tried loading from a memory card save instead of a state, and I also experienced these crashes early in the game when I was trying to only use the memory card and no save states (trying to be "authentic", I guess :-)).  I've gotten into the habit of saving state, saving to memory card, and saving state again before exiting out, and then I typically load from state because it's faster.  But I periodically load from the memory card save and create a new state when there is an update to avoid the possibility of a corrupted state from a past version (assuming the mempry card save isn't messed up in some way).  It has been a while since I tried starting a new game with a clean slate tp make sure it still occurs there.

**Comment by gnafuthegreat** [comment]
An update: I just finished my first play-through of Chrono Trigger tonight, so I won't actually be playing it much (at least for a while; I may yet come back to it to try and get some different endings).  I'm sorry I waited so long to file this bug!

Anyway, I'm happy to continue to test if anyone has any suggestions or other logs I cam collect.  My main concern is that the crashing game can bring down RetroArch altogether, and that this causes the memory card file to not get updated with recemt saves.  The crashing wouldn't be so annoying if it weren't for the lost save data.  Not sure how much can be done vs. what's handled by the BIOS and beyond your control.
